### ISSUE TITLE
(URGENT) Ribose PDF: Duplicated "IETF" keyword in non-fetched bibliographic reference

### ISSUE THREAD

**Issue by ronaldtse** [metanorma/metanorma-ribose#104]
```
* [[[IETF_BIBXML_RFP,IETF BIBXML RFP]]], _BibXML Service RFP_.
```

Rendered as: 
<img width="367" alt="Screenshot 2021-05-31 at 8 40 02 PM" src="https://user-images.githubusercontent.com/11865/120194631-6ace2880-c250-11eb-8377-652789ed4e44.png">

**Comment by ronaldtse** [comment]
Or is this a PDF error?

**Comment by opoudjis** [comment]
IETF is prefixed regularly to all IETF references: _RFC 1000_ becomes _IETF RFC 1000, I.-D_. Bogus becomes _IETF I.-D. Bogus_. This rule is only sidestepped in IETF documents themselves.

If all of a sudden a document I have never heard of already has "IETF" in its document identifier, then that document is simply malformed. It is inconsistent with the other documents coming out of IETF. And I am not a mindreader.

The thing that's confusing here is, there is code already in place to prevent repeating an existing prefix, but it is not being invoked. Investigating.

**Comment by opoudjis** [comment]
Cannot duplicate this:

<img width="977" alt="Στιγμιότυπο 2021-06-01, 00 25 32" src="https://user-images.githubusercontent.com/1911039/120207607-fd31f480-c26f-11eb-9cd8-9c4640c94385.png">
<img width="1252" alt="Στιγμιότυπο 2021-06-01, 00 25 05" src="https://user-images.githubusercontent.com/1911039/120207620-01f6a880-c270-11eb-905b-b2663cf253f3.png">
<img width="1238" alt="Στιγμιότυπο 2021-06-01, 00 24 56" src="https://user-images.githubusercontent.com/1911039/120207624-0327d580-c270-11eb-9307-a748a33b4420.png">

**Comment by ronaldtse** [comment]
This is from normative references. I'm using metanorma-ribose git gem.

<img width="515" alt="Screenshot 2021-05-31 at 10 47 19 PM" src="https://user-images.githubusercontent.com/11865/120210290-33687780-c262-11eb-997e-6a0ebe251fb3.png">

**Comment by ronaldtse** [comment]
@opoudjis verified -- it happens in Normative references, does not happen in Bibliography.

**Comment by ronaldtse** [comment]
Notice I'm also seeing "ISO ISO 690".

**Comment by ronaldtse** [comment]
@opoudjis this is not an XML problem, I think this is a PDF problem: `<docidentifier type="IETF">IETF BIBXML RFP</docidentifier>`.

@Intelligent2013 can you help here?

**Comment by ronaldtse** [comment]
@Intelligent2013 this issue only occurs in "Normative references", the "docidentifier type" is being shown.

**Comment by Intelligent2013** [comment]
> 
> 
> @Intelligent2013 this issue only occurs in "Normative references", the "docidentifier type" is being shown.

In xslt bibitem rendering depends on parent section.
For 'Normative reference':
```xml
<xsl:if test="rsd:docidentifier">
	<xsl:if test="rsd:docidentifier/@type != 'IETF' and rsd:docidentifier/@type != 'ISO'">
		<xsl:value-of select="rsd:docidentifier/@type"/>
		<xsl:text> </xsl:text>
	</xsl:if>
	<xsl:value-of select="rsd:docidentifier"/>
</xsl:if>
<xsl:apply-templates select="rsd:note"/>
<xsl:if test="rsd:docidentifier">, </xsl:if>
<xsl:choose>
	<xsl:when test="rsd:formattedref">
		<xsl:apply-templates select="rsd:formattedref"/>
	</xsl:when>
	<xsl:otherwise>
		<xsl:for-each select="rsd:contributor[rsd:role/@type='publisher']/rsd:organization/rsd:name">
			<xsl:apply-templates/>
			<xsl:if test="position() != last()">, </xsl:if>
			<xsl:if test="position() = last()">: </xsl:if>
		</xsl:for-each>
		<!-- rsd:docidentifier -->
		<fo:inline font-style="italic">
			<xsl:choose>
				<xsl:when test="rsd:title[@type = 'main' and @language = 'en']">
					<xsl:value-of select="rsd:title[@type = 'main' and @language = 'en']"/>
					<xsl:text>. </xsl:text>
				</xsl:when>
				<xsl:otherwise>
					<xsl:value-of select="rsd:title"/>
					<xsl:text>. </xsl:text>
				</xsl:otherwise>
			</xsl:choose>
		</fo:inline>
		<xsl:for-each select="rsd:contributor[rsd:role/@type='publisher']/rsd:organization/rsd:name">
			<xsl:apply-templates/>
			<xsl:if test="position() != last()">, </xsl:if>
		</xsl:for-each>
		<xsl:if test="rsd:date[@type='published']/rsd:on">
			<xsl:text> (</xsl:text>
			<xsl:value-of select="rsd:date[@type='published']/rsd:on"/>
			<xsl:text>)</xsl:text>
		</xsl:if>
	</xsl:otherwise>
</xsl:choose>
```
For Bibliography section:
```xml
<xsl:if test="rsd:docidentifier">
	<xsl:choose>
		<xsl:when test="rsd:docidentifier/@type = 'metanorma'"/>
		<xsl:otherwise>
			<fo:inline>
				<xsl:value-of select="rsd:docidentifier"/>, </fo:inline>
		</xsl:otherwise>
	</xsl:choose>
</xsl:if>
<xsl:choose>
	<xsl:when test="rsd:title[@type = 'main' and @language = 'en']">
		<xsl:apply-templates select="rsd:title[@type = 'main' and @language = 'en']"/>
	</xsl:when>
	<xsl:otherwise>
		<xsl:apply-templates select="rsd:title"/>
	</xsl:otherwise>
</xsl:choose>
<xsl:apply-templates select="rsd:formattedref"/>
```

**Comment by Intelligent2013** [comment]
docidentifier type disabled for @type=IETF and ISO.

**Comment by ronaldtse** [comment]
This will only be fully resolved when #114 is fixed.
